BackEnd/src/test/kotlin/quest/marketstack/tradingapp/opentrades/controller/OpenOpenTradeControllerTest.kt

```diff
@@ -0,0 +1,348 @@
+package quest.marketstack.tradingapp.opentrades.controller
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.hamcrest.Matchers.hasSize
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType
+import org.springframework.test.annotation.DirtiesContext
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.get
+import org.springframework.test.web.servlet.post
+import quest.marketstack.tradingapp.model.TradeExec
+import java.time.LocalDate
+import java.time.LocalTime
+
+@ActiveProfiles("test")
+@SpringBootTest
+@AutoConfigureMockMvc
+class OpenOpenTradeControllerTest
+    @Autowired
+    constructor(
+        val mockMvc: MockMvc,
+        val objectMapper: ObjectMapper,
+    ) {
+        @Value("\${api.open-trade.base-url}")
```

Comment:
This is going to get deleted anyway, but as an example, see how I referenced application-test.yml here for the baseUrl? See how doing it in the yml is useful? Imagine if you had 20 tests in 5 different files with the URL hardcoded, would need to change it everywhere. Now you only need to change it in one place.